components/ai_chat/resources/page/components/filter_menu/tools_menu.test.tsx

```diff
@@ -60,41 +89,164 @@ describe('tools_menu', () => {
     query: null,
   }
 
-  it('should render', () => {
-    const { getByText, container } = render(<ToolsMenu {...defaultProps} />)
-    expect(getByText('category 1')).toBeInTheDocument()
-    expect(getByText('category 2')).toBeInTheDocument()
-    expect(getByText('subheading 1')).toBeInTheDocument()
-    expect(getByText('subheading 2')).toBeInTheDocument()
-    expect(getByText('entry 1')).toBeInTheDocument()
-    expect(getByText('entry 2')).toBeInTheDocument()
-    expect(getByText('entry 3')).toBeInTheDocument()
-
-    expect(container.querySelector('leo-buttonmenu')).toHaveProperty(
-      'isOpen',
-      true,
+  const renderWithContext = (
+    smartModes: Mojom.SmartMode[] = mockSmartModes,
+    props = defaultProps,
+  ) => {
+    const aiChatContext = { smartModes }
+    return render(
+      <AIChatReactContext.Provider value={aiChatContext as any}>
+        <ToolsMenu {...props} />
+      </AIChatReactContext.Provider>,
     )
+  }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
   })
 
-  it('should hide subheadins when filtering', () => {
-    const { queryByText } = render(
-      <ToolsMenu
-        {...defaultProps}
-        query='e'
-      />,
-    )
-    expect(queryByText('subheading 1')).not.toBeInTheDocument()
-    expect(queryByText('subheading 2')).not.toBeInTheDocument()
+  describe('without smart modes', () => {
+    it('should render categories without smart modes', () => {
```

Comment:
this is just checking that the smart modes category is hidden when there are no smart modes right?

Maybe 
```suggestion
    it('smart modes category is hidden when there are no smart modes', () => {
```